### ISSUE TITLE
Screen readers can see existing reactions but not their senders

### ISSUE THREAD

**Issue by t3chguy** [vector-im/riot-web#11608]

**Comment by MarcoZehe** [comment]
The buttons for reactions aren't actually buttons currently, and the tooltips only appear on hover and are not accessible with correct roles. There are two possibilities:

1. Simply take what would be displayed in the tool tip and add that textual information to the button as the label. That way, screen reader users would not need to bring up the tool tips.
   * Advantage: Information would be readily available, just would need to be formatted in a way that includes the number of reactions for each, and who reacted.
   * Disadvantage. The information for each tool tip would have to be processed twice, once for the label for the screen reader users, once when the tool tip is generated.
2. Add proper accessibility to the tool tip, make the buttons accessible, and associate the tooltip to the button via aria-describedby.
   * Advantage: The experience is closer to what sighted people get.
   * Advantage: The label is only generated once.
   * Disadvantage: For each tooltip, an ID reference needs to be generated so aria-describedby on the associated reaction button can reference it and screen readers can speak it on additional information.
   * Disadvantage: aria-describedby is only rendered on focus, not inside the virtual buffer in many circumstances. Some browsers also have bugs if the item that aria-describedby references must not be hidden, or the association won't work.

I'd personally prefer solution 1 for more conciseness and less hassle and inconsistency with different browser/screen reader combinations, but realize that this is a bit further away from the experience of sighted users. Screen reader users would get more information for each button upfront.
